### PR TITLE
Update web request examples

### PR DESCRIPTION
**Pull request by pollosp**
Hello, 
New guy using Rust here. 
This quarantine morning I was trying to run the current example of simple request. I just was expecting  to copy and paste it and see how it work, but first thing I got was:

```
$ cargo build
   Compiling tokio v0.2.18
   Compiling tokio-util v0.3.1
   Compiling tokio-tls v0.3.0
   Compiling h2 v0.2.4
   Compiling hyper v0.13.5
   Compiling hyper-tls v0.4.1
   Compiling reqwest v0.10.4
   Compiling restget v0.1.0 (/Users/omar/code/rust/restget)
error[E0107]: wrong number of type arguments: expected 2, found 1
 --> src/main.rs:5:14
  |
5 | fn main() -> Result<()> {
  |              ^^^^^^^^^^ expected 2 type arguments

```
With the help of @onielfa he manage to fix the example , there is something we are not catching form the original example or it wasn't working with current version of rust:
```
$ rustc --version
rustc 1.42.0 (b8cedc004 2020-03-09)
```

and modules: 

```
reqwest = { version = "0.10"}
tokio = { version = "0.2", features = ["full"] }
```
 
Also will be nice to have the `Cargo.toml` requirements into the documentation, specially for people which is starting with Rust and they go directly to the example.

I don't know if this is OK or NOT , but maybe could help.

### CODE REVIEW

**Comment by AndyGauge** [comment]
Nice, very similar to what I'm introducing in #575 

What I find is when I update the reqwest version to .10, 15 recipes break.  I can't accept a PR that breaks 14 other examples.

The workaround is to use Tokio 0.8 as show inside the Cargo.toml file (but isn't spelled out in every recipe).  The solution is to update 15 recipes.

**Comment by pollosp** [comment]
Oh I see , maybe I could try thanks for the feedback @AndyGauge and I didn't see #575 🙀 sorry!

**Comment by AndyGauge** [comment]
Id be happy to accept a PR that addresses all the recipes.  I'm unlikely to finish #575 on my own.

**Comment by pollosp** [comment]
@AndyGauge I make work some code into the examples but I don't know why test are still breaking, I was checking https://github.com/budziq/rust-skeptic but for me everything is ok :S, any clue why the code is still breaking? As I told you I'm new with rust in fact this is my second day 🙀 

Thanks

File: src/web/scraping/extract-links.md [hunk 1]
```diff
@@ -9,31 +9,25 @@ Call [`filter_map`] on the [`Selection`] retrieves URLs
...
+    let res = reqwest::get("https://www.rust-lang.org/en-US/").await?.text().await?;
+
+    Document::from(res.as_str())
         .find(Name("a"))
         .filter_map(|n| n.attr("href"))
         .for_each(|x| println!("{}", x));
 
     Ok(())
 }
+}
```
Comment (AndyGauge) [inline src/web/scraping/extract-links.md:30]: This is causing a failure
error: unexpected closing delimiter: `}`
  --> /tmp/rust-skeptic.v9NBrzp3zOUB/test.rs:20:1
   |
20 | }
   | ^ unexpected closing delimiter
error: aborting due to previous error
test extract_links_sect_extract_all_links_from_a_webpage_html_line_11 ... FAILED

File: src/web/scraping/extract-links.md [hunk 2]
```diff
@@ -9,31 +9,25 @@ Call [`filter_map`] on the [`Selection`] retrieves URLs
...
-#    foreign_links {
-#        ReqError(reqwest::Error);
-#        IoError(std::io::Error);
-#    }
-# }
-
-fn main() -> Result<()> {
-    let res = reqwest::get("https://www.rust-lang.org/en-US/")?;
-
-    Document::from_read(res)?
```
Comment (AndyGauge) [inline src/web/scraping/extract-links.md:30]: The proper error handling here is something we want to highlight, not remove

File: src/web/scraping/unique.md [hunk 3]
```diff
@@ -9,27 +9,20 @@ look for all entries of internal and external links with
...
-#     foreign_links {
-#         Io(std::io::Error);
-#         Reqwest(reqwest::Error);
-#         Regex(regex::Error);
-#     }
-# }
-#
-fn extract_links(content: &str) -> Result<HashSet<Cow<str>>> {
+
+#[tokio::main]
```
Comment (AndyGauge) [inline src/web/scraping/unique.md:23]: Doesn't this only work right above the main function?

File: src/web/scraping/unique.md [hunk 4]
```diff
@@ -9,27 +9,20 @@ look for all entries of internal and external links with
...
+
+async fn extract_links(content: &str) -> HashSet<Cow<str>> {
```
Comment (AndyGauge) [inline src/web/scraping/unique.md:25]: what about this function is async?

File: src/web/clients/requests/get.md [hunk 5]
```diff
@@ -8,32 +8,25 @@ status and headers. Reads HTTP response body into an allocated [`String`]
...
     println!("Status: {}", res.status());
     println!("Headers:\n{:#?}", res.headers());
+    let body = res.text().await?;
     println!("Body:\n{}", body);
-
     Ok(())
 }
 ```
 
+`Cargo.toml`
```
Comment (AndyGauge) [inline src/web/clients/requests/get.md:23]: It is an interesting design decision but the cookbook hasn't endorsed it.

File: src/web/clients/requests/get.md [hunk 6]
```diff
@@ -8,32 +8,25 @@ status and headers. Reads HTTP response body into an allocated [`String`]
...
-#     }
-# }
-
-fn main() -> Result<()> {
-    let mut res = reqwest::get("http://httpbin.org/get")?;
-    let mut body = String::new();
-    res.read_to_string(&mut body)?;
-
+#[tokio::main]
+async fn main() -> Result<(), reqwest::Error> {
```
Comment (AndyGauge) [inline src/web/clients/requests/get.md:13]: error[E0670]: `async fn` is not permitted in the 2015 edition
 --> /tmp/rust-skeptic.rEEf54DF9Cer/test.rs:4:1
  |
4 | async fn main() -> Result<(), reqwest::Error> {
  | ^^^^^
error[E0609]: no field `await` on type `impl std::future::Future`
 --> /tmp/rust-skeptic.rEEf54DF9Cer/test.rs:5:54
  |
5 |     let res = reqwest::get("http://httpbin.org/get").await?;
  |                                                      ^^^^^
  |
  = note: to `.await` a `Future`, switch to Rust 2018
  = help: set `edition = "2018"` in `Cargo.toml`
  = note: for more on editions, read https://doc.rust-lang.org/edition-guide
error: aborting due to 2 previous errors
Some errors have detailed explanations: E0609, E0670.
For more information about an error, try `rustc --explain E0609`.
test get_sect_make_a_http_get_request_line_10 ... FAILED

Comment (AndyGauge) [reply src/web/clients/requests/get.md:13]: Looks like this work is blocked by #530 which ideally is a change in error-chain to push through the PR.  You should be able to add edition2018 to the infostring:
\```rust,edition2018

File: src/web/clients/api/rest-post.md [hunk 7]
```diff
@@ -12,32 +12,28 @@ body. [`RequestBuilder::basic_auth`] handles authentication. The call to
...
     id: String,
     html_url: String,
 }
 
-fn main() -> Result<()> {
-    let gh_user = env::var("GH_USER")?;
+#[tokio::main]
+
+async fn main() ->  Result<(),Box<dyn Error>> {
+    let gh_user = env::var("we")?;
```
Comment (AndyGauge) [inline src/web/clients/api/rest-post.md:36]: why the change in environment variable?

**Comment by AndyGauge** [comment]
The scope is a bit extreme.  Maybe we should concentrate on one recipe at a time.

File: src/web/scraping/unique.md (hunk 3 above)
Comment (pollosp) [reply src/web/scraping/unique.md:23]: Nop I tried it , because i expected that and not :S

File: src/web/clients/api/rest-post.md (hunk 7 above)
Comment (pollosp) [reply src/web/clients/api/rest-post.md:36]: Ups typo pushing

**Comment by pollosp** [comment]
Looks like there are less test to fix:
I started with 13:
```
    basic_sect_download_a_file_to_a_temporary_directory_line_12
    broken_sect_check_a_webpage_for_broken_links_line_13
    extract_links_sect_extract_all_links_from_a_webpage_html_line_11
    get_sect_make_a_http_get_request_line_10
    paginated_sect_consume_a_paginated_restful_api_line_9
    partial_sect_make_a_partial_download_with_http_range_headers_line_12
    post_file_sect_post_a_file_to_paste_rs_line_12
    request_sect_parse_the_mime_type_of_a_http_response_line_14
    rest_get_sect_query_the_github_api_line_8
    rest_head_sect_check_if_an_api_resource_exists_line_11
    rest_post_sect_create_and_delete_gist_with_github_api_line_14
    retain_sect_avoid_discarding_errors_during_error_conversions_line_15
    unique_sect_extract_all_unique_links_from_a_mediawiki_markup_line_11
```

and right now are 7 : 

```
    basic_sect_download_a_file_to_a_temporary_directory_line_12
    broken_sect_check_a_webpage_for_broken_links_line_13
    paginated_sect_consume_a_paginated_restful_api_line_9
    partial_sect_make_a_partial_download_with_http_range_headers_line_12
    post_file_sect_post_a_file_to_paste_rs_line_12
    request_sect_parse_the_mime_type_of_a_http_response_line_14
    retain_sect_avoid_discarding_errors_during_error_conversions_line_15
```

Was hard to me to read the test output :S , right now is more clear thanks for your examples @AndyGauge and support. 

I will try to follow on this during the next weekend

File: src/web/clients/requests/get.md (hunk 6 above)
Comment (pollosp) [reply src/web/clients/requests/get.md:13]: rust,edition2018 doesn't work to me :S , setting it at the `Cargo.toml` as error report didn't break any other test.

**Comment by pollosp** [comment]
Just 3 left
```
    broken_sect_check_a_webpage_for_broken_links_line_13
    paginated_sect_consume_a_paginated_restful_api_line_9
    retain_sect_avoid_discarding_errors_during_error_conversions_line_15
```

**Comment by pollosp** [comment]
Two remaining:
```
 paginated_sect_consume_a_paginated_restful_api_line_9
 retain_sect_avoid_discarding_errors_during_error_conversions_line_15
```
I'm having troubles with the error types there , we will see 🧷

**Comment by AndyGauge** [comment]
Your work made it into the cookbook.  Thank you for your contribution!
